scaffolding-node/lib/scaffolding.sh

```diff
@@ -422,19 +423,28 @@ _detect_git() {
 }
 
 _detect_node() {
+
   if [[ -n "$scaffolding_node_pkg" ]]; then
     _node_pkg="$scaffolding_node_pkg"
     build_line "Detected Node.js version in Plan, using '$_node_pkg'"
   else
     local val
     val="$(_json_val package.json .engines.node)"
     if [[ -n "$val" ]]; then
+
       # TODO fin: Add more robust packages.json to Habitat package matching
       case "$val" in
         *)
-          local version_num
-          version_num=$(_extracted_version_number "$val")
-          _node_pkg="core/node/$version_num"
+          nearest_version_string=$(_nearest_version_on_builder "$val")
+
+#          nearest_version=$([[ $nearest_version_string =~ ([0-9]\.[0-9]\.[0-9]) ]] && echo "${BASH_REMATCH[1]}")
+          nearest_version="${nearest_version_string//\"}"
+
+          if [[ $nearest_version =~ (.*No compatible version of node found.*) ]]; then
```

Comment:
Is this where we would fail a build? If so, you could use the `exit_with()` function which kills the build program with a non-zero exit code. Like [this](https://github.com/habitat-sh/core-plans/blob/b5f1805244cd478c9047512676f75ef336593aae/scaffolding-node/lib/scaffolding.sh#L256-L258)